### ISSUE TITLE
Stop opening folders with vscode

### ISSUE THREAD

**Issue by guest73** [microsoft/vscode#80499]
Please, when installing vscode on linux machine DO NOT register it for inode/directory mime type. Each time I want "Show in folder" in Chromium/Chrome/Opera the vscode is opened with the folder's content. And I cannot find a way to stop this behavior. It drives me nuts! Who even came up with an idea to make it like this?!! File managers exist for this purposes.  All I want with vscode is writing the damn code, not opening folders.

**Comment by joaomoreno** [comment]
Might this be an issue with your desktop environment maybe? We do not register for the directory mimetype afaik.

**Comment by guest73** [comment]
Might be... I do not know.
I recently found a solution which worked for me https://bugs.chromium.org/p/chromium/issues/detail?id=918870#c11.
But who puts those lines into /etc/mailcap? And why does Mozilla obey xdg-related defaults but all chrome-like browsers are rather looking at /etc/mailcap by default? Not the questions to be asked here though...
Anyway, you say "We do not register for the directory mimetype afaik." but I installed the vscode from an additional repo
 deb [arch=amd64] http://packages.microsoft.com/repos/vscode stable main
which seems to be an official stuff. The instructions for default-apps/mime-types must be in the .deb file from the repo (do not have must knowledge on the topic, but it's logical. The desktop environment just can't magically pick one apps for folder-opening and ignore others). And someone made the deb.

I made this issue because I was mad at this situation not being able to open a folder with a downloaded file from chrome in my file manager because vscode started instead. And I still have a strong opinion that IDE's should not be registered as default apps for folder opening. But I guess someone has a different opinion and it's been done this way intentionally.

To sum it up. @joaomoreno , it's up to you whether to close the issue or do something with it. For now I am happy as I somehow solved my problem.

**Comment by joaomoreno** [comment]
You know, I think you are right after all: https://github.com/microsoft/vscode/blob/c05a0af7dba62a49eee4bb34b6f666f68a9812a5/resources/linux/code.desktop#L11

This was added in https://github.com/microsoft/vscode/commit/23217e2f4bffc9f06a2de51a8d82a3d87223468a to fix https://github.com/microsoft/vscode/issues/15741

@Tyriar might there be another way of fixing https://github.com/microsoft/vscode/issues/15741?

**Comment by Tyriar** [comment]
> And I still have a strong opinion that IDE's should not be registered as default apps for folder opening.

We don't do this, what I got from looking into this in the past was that some desktop environments are misbehaving and should have set the file manager to the default like it does under Ubuntu. We're just saying vscode can be opened, not that it should be a default.

@guest73 if you have a suggestion for how we are registering to open directory incorrectly then we can fix it. Regardless, you should be able to change the default using xdg utils.

**Comment by guest73** [comment]
Well, ok. I've told you I am fine with the current solution I found out. But just to clarify. I am not using a desktop environment it's rather a simple i3wm window manager with a few xfce4 services. And xdg utils do not help as Chrome, seems, does not respect those on environment that is now a full featured Desktop environment. FF does respect it though.
 Anyway, thank for your time guys!
